pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -0,0 +1,304 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from enum import Enum
+from decimal import Decimal
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, truncated_discrete_set, \
+    joined_validators, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Trace(str, Enum):
+    A = "TRA"
+    B = "TRB"
+
+
+class MixerMode(str, Enum):
+    Internal = "INT"
+    External = "EXT"
+
+
+class CouplingMode(str, Enum):
+    AC = "AC"
+    DC = "DC"
+
+
+class DemodulationMode(str, Enum):
+    Amplitude = "AM"
+    Frequency = "FM"
+    Off = "OFF"
+
+
+class DetectionModes(str, Enum):
+    NegativePeak = "NEG"
+    Normal = "NRM"
+    PositivePeak = "POS"
+    Sample = "SMP"
+
+
+class AmplitudeUnits(str, Enum):
+    DBM = "DBM"
+    DBMV = "DBMV"
+    DBUV = "DBUV"
+    V = "V"
+    W = "W"
+    AUTO = "AUTO"
+    MANUAL = "MAN"
+
+
+class HP856Xx(Instrument):
+    """
+
+    Some command descriptions are taken from the document: 'HP 8560A, 8561B Operating & Programming'
+    """
+
+    def __init__(self, adapter, name="Hewlett-Packard HP856Xx", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            send_end=True,
+            **kwargs,
+        )
+
+    id = Instrument.measurement(
+        "ID?",
+        """
+        Get identification of the device with software and hardware revision (e.g. HP8560A,002,H03)
+        """,
+        maxsplit=0
+    )
+
+    attenuation = Instrument.control(
+        "AT?", "AT %s",
+        """
+        Controls input attenuation in decade steps from 10 to 70db or set to AUTO and MAN(ual)
```

Comment:
Docsreings should be written in the impterative form, in this case "control" 
Please use a space between the value and the unit desination (dB)